mars/dataframe/tests/test_core.py

```diff
@@ -320,3 +320,90 @@ def test_key_value(setup):
 
     result = idx.values.execute().fetch()
     np.testing.assert_array_equal(result, raw.values)
+
+
+def test_between(setup):
+    series = Series(pd.date_range("1/1/2000", periods=10))
+    left, right = series[2], series[7]
+
+    result = series.between(left, right).execute().fetch()
+    expected = (series >= left) & (series <= right)
+    expected = expected.execute().fetch()
+    pd.testing.assert_series_equal(result, expected)
+
+
+def test_between_datetime_values(setup):
+    ser = Series(pd.bdate_range("1/1/2000", periods=20).astype(object))
+    ser = ser.execute().fetch()
+    ser[::2] = np.nan
+
+    result = ser[ser.between(ser[3], ser[17])]
+    expected = ser[3:18].dropna()
+    pd.testing.assert_series_equal(result, expected)
+
+    result = ser[ser.between(ser[3], ser[17], inclusive="neither")]
+    expected = ser[5:16].dropna()
+    pd.testing.assert_series_equal(result, expected)
+
+
+def test_between_period_values(setup):
+    ser = Series(pd.period_range("2000-01-01", periods=10, freq="D"))
+    left, right = ser[2], ser[7]
+    result = ser.between(left, right).execute().fetch()
+    expected = (ser >= left) & (ser <= right)
+    expected = expected.execute().fetch()
+    pd.testing.assert_series_equal(result, expected)
+
+
+def test_between_inclusive_string(setup):
+    series = Series(pd.date_range("1/1/2000", periods=10))
```

Comment:
The test can be like.

```
pd_series = pd.xxx
left, right = pd_series[2], pd_series[7]
series = Series(pd_series)
...
result = series.between).to_pandas()
expected = pd_series.between(left, right)
```

The expected part can use pandas directly, the result part in Mars.